appinventor/appengine/src/com/google/appinventor/client/boxes/PaletteBox.java

```diff
@@ -38,5 +39,9 @@ private PaletteBox() {
         false,     // startMinimized
         false,     // usePadding
         false);    // highlightCaption
+
+    // UIBinder calls the private constructor, presumably through reflection, which
```

Comment:
> I don't understand this comment because UiBinder cannot call private constructors. The Java code is generated and so still follows all of the standard Java visibility rules.

I'm not sure what the justification is, but objects declared in UiBinder that do not have public no-parameter constructors do seem to be able to call the private constructor. This was leading to two instances floating around of classes that were supposed to be singletons.

I found a better way of handling this for most cases. I'm not sure if this is one I forgot to update or if there's a reason it didn't work. Looking at it.

Reply:
OK, this should be resolved.